packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -254,69 +254,24 @@ function parseElementDirectives(
     let current: ParentNode | undefined;
 
     const parsers = [
-        parseForEach,
-        parseForOf,
-        parseIf,
         parseIfBlock,
         parseElseifBlock,
         parseElseBlock,
+        parseForEach,
+        parseForOf,
+        parseIf,
     ];
     for (const parser of parsers) {
```

Comment:
For this loop, I refactored the logic that adds the node to the current scope and attaches them to the appropriate parent node to the actual parsing functions for each directive. The reasoning is twofold:

1. half the directives now have "special" logic on what to do when attaching them, so it's no longer very streamlined to do it here. 
2. there's no longer a 1:1 mapping between Node type and parser function here, so distinguishing what to do with with a given node at this level is trickier. 2 of the 6 nodes here would need some combination of `ast.isXXXXBlock` and checking for function equality to determine what to do with them. We already have all that info readily available in the parsing functions.